### PR TITLE
feat(notebook): Add survey node

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

Adds a basic survey node to notebooks

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Review by liyiy — APPROVED** [review]

File: frontend/src/scenes/surveys/surveyLogic.tsx [hunk 1]
```diff
@@ -153,9 +86,9 @@ export interface SurveyMetricsQueries {
 }
 
 export const surveyLogic = kea<surveyLogicType>([
-    path(['scenes', 'surveys', 'surveyLogic']),
     props({} as SurveyLogicProps),
     key(({ id }) => id),
+    path((key) => ['scenes', 'surveys', 'surveyLogic', key]),
```
Comment (neilkakkar) [inline frontend/src/scenes/surveys/surveyLogic.tsx:91]: @liyiy mind testing once more if everything is working fine for survey results? I did, and it seemed fine to me.

I think you had to create all those setters because `key` was missing from the path, hence why selectors weren't updating, because they were all on the same logic

File: frontend/src/scenes/notebooks/Nodes/NotebookNodeSurvey.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        <div>
+            <BindLogic logic={surveyLogic} props={{ id }}>
+                <div className="flex items-center gap-2 p-3">
+                    <IconSurveys className="text-lg" />
+                    {surveyLoading ? (
+                        <LemonSkeleton className="h-6 flex-1" />
+                    ) : (
+                        <>
+                            <span className="flex-1 font-semibold truncate">{survey.name}</span>
+                            {/* survey has to exist in notebooks */}
```
Comment (daibhin) [inline frontend/src/scenes/notebooks/Nodes/NotebookNodeSurvey.tsx:34]: Might be worth adding an error state in cases when the survey is missing. To prevent the need for casting or an optional object

File: frontend/src/scenes/notebooks/Nodes/NotebookNodeSurvey.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+                <LemonDivider className="my-0" />
+                <div className="p-2 mr-1 flex justify-end gap-2">
+                    {survey.linked_flag && (
+                        <LemonButton
+                            type="secondary"
+                            size="small"
+                            icon={<IconFlag />}
+                            onClick={(e) => {
+                                e.stopPropagation()
```
Comment (daibhin) [inline frontend/src/scenes/notebooks/Nodes/NotebookNodeSurvey.tsx:92]: Why was this needed again? Ideally we should come up with a better way that doesn't force you to have remember to do this

File: frontend/src/scenes/notebooks/Nodes/NotebookNodeSurvey.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+                                if (nextNode?.type.name !== NotebookNodeType.FeatureFlag) {
+                                    insertAfter(buildFlagContent((survey.linked_flag as FeatureFlagBasicType).id))
+                                }
+                            }}
+                            disabledReason={
```
Comment (daibhin) [inline frontend/src/scenes/notebooks/Nodes/NotebookNodeSurvey.tsx:98]: Maybe instead of disabling the button this could set the flag node to be active?

**Review by daibhin — APPROVED** [review]
Looks good. I mainly reviewed the notebooks related changes but took a pass at the other parts too

File: frontend/src/scenes/notebooks/Nodes/NotebookNodeSurvey.tsx (hunk 3 above)
Comment (neilkakkar) [reply frontend/src/scenes/notebooks/Nodes/NotebookNodeSurvey.tsx:92]: So when I click on this action, it doesn't expand the node as well, which also happens on click anywhere inside the component.

File: frontend/src/scenes/notebooks/Nodes/NotebookNodeSurvey.tsx (hunk 4 above)
Comment (neilkakkar) [reply frontend/src/scenes/notebooks/Nodes/NotebookNodeSurvey.tsx:98]: oooh good idea, I like this! Will do in a follow up for everything
